File: 프로그래머스/level2/43165.py
answer = 0

def make_number(k, arr, tmp, target):
    global answer
    if k == len(arr):
        if tmp == target:
            answer += 1; return
        else: return
    make_number(k + 1, arr, tmp + arr[k], target)
    make_number(k + 1, arr, tmp - arr[k], target)

def solution(numbers, target):
    global answer
    make_number(0, numbers, 0, target)
    return answer

print(solution([1, 1, 1, 1, 1], 3))

File: 프로그래머스/level2/43165.js
let answer = 0

const calc = function(k, numbers, target, tmp) {
    if (k == numbers.length) {
        if (tmp === target) {
            console.log('hi')
            answer += 1
            return
        } else return
    }
    calc(k + 1, numbers, target, tmp + numbers[k])
    calc(k + 1, numbers, target, tmp - numbers[k])
}

const solution = (numbers, target) => {
    calc(0, numbers, target, 0)
    return answer
}

console.log(solution([1, 1, 1, 1, 1], 3))